src/umpire/strategy/DynamicPool.cpp

```diff
@@ -21,374 +21,277 @@
 #include "umpire/Replay.hpp"
 
 #include <algorithm>
+#include <tuple>
+
+static std::size_t round_up(std::size_t num, std::size_t factor)
+{
+  return num + factor - 1 - (num - 1) % factor;
+}
 
 namespace umpire {
 namespace strategy {
 
-
-void DynamicPool::findUsableBlock(struct Block *&best, struct Block *&prev, std::size_t size) {
-  best = prev = nullptr;
-  for ( struct Block *iter = freeBlocks, *iterPrev = nullptr ; iter ; iter = iter->next ) {
-    if ( iter->size >= size && (!best || iter->size < best->size) ) {
-      best = iter;
-      prev = iterPrev;
-      if ( iter->size == size )
-        break;    // Exact match, won't find a better one, look no further
-    }
-    iterPrev = iter;
-  }
-}
-
-inline static std::size_t alignmentAdjust(const std::size_t size) {
-  const std::size_t AlignmentBoundary = 16;
-  return std::size_t (size + (AlignmentBoundary-1)) & ~(AlignmentBoundary-1);
+DynamicPool::DynamicPool(const std::string& name,
+                         int id,
+                         Allocator allocator,
+                         const std::size_t initial_alloc_bytes,
+                         const std::size_t min_alloc_bytes,
+                         const int align_bytes,
+                         Coalesce_Heuristic coalesce_heuristic) noexcept :
+  AllocationStrategy(name, id),
+  m_allocator{allocator.getAllocationStrategy()},
+  m_min_alloc_bytes{min_alloc_bytes},
+  m_align_bytes{align_bytes},
+  m_do_coalesce{coalesce_heuristic},
+  m_used_map{},
+  m_free_map{},
+  m_curr_bytes{0},
+  m_actual_bytes{initial_alloc_bytes},
+  m_highwatermark{initial_alloc_bytes}
+{
+  insertFree(m_allocator->allocate(initial_alloc_bytes), initial_alloc_bytes, true);
```

Comment:
This version also allocates the `initial_alloc_bytes` block in the constructor.

Reply:
This is a good change, some users have been confused by the previous behavior.

Reply:
If I understand correctly, the only thing that changed here is the name from `min_initial_alloc_size` to `initial_alloc_bytes`.  If this is so, the only change was the name (which is good), but not the signature.  The only signature change was the addition of the `align_bytes` optional argument at the end.

Please confirm that I understand this correctly.

Reply:
The `align_bytes` parameter was added to the signature of the constructor. This new version allocates a free block in the constructor, whereas the version on develop allocates that the first time an allocation is requested.

Reply:
Aha!  Yes!! That is much better!  Thanks for clarifying that for me.